File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/reactiveobject/ReadHistoryDataResponse.java
package org.opcfoundation.webserver.service.message.reactiveobject;

import org.eclipse.milo.opcua.stack.core.types.builtin.DataValue;

import java.util.List;

public class ReadHistoryDataResponse {
    private final List<DataValue> values;
    private final Boolean containsMoreData;

    public ReadHistoryDataResponse(List<DataValue> values, Boolean containsMoreData)
    {
        this.values = values;
        this.containsMoreData = containsMoreData;
    }

    public List<DataValue> getValues() {
        return values;
    }

    public Boolean getContainsMoreData() {
        return containsMoreData;
    }
}

File: opcua-webservice-node/source/src/server/service/message/reactiveobject/ReadHistoryDataResponse.ts
import { UaDataValue } from "opcua-webapi-ts";

export class ReadHistoryDataResponse {
    private readonly _values: Array<UaDataValue>;
    private readonly _containsMoreData: boolean;

    constructor(values: Array<UaDataValue>, containsMoreData: boolean) {
        this._values = values;
        this._containsMoreData = containsMoreData;
    }

    get values(): Array<UaDataValue> {
        return this._values;
    }

    get containsMoreData(): boolean {
        return this._containsMoreData;
    }
}